src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
+	itm, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating mail message")
+	}
+
+	if itm == nil {
+		return nil, clues.New("nil response mail message creation").WithClues(ctx)
+	}
+
+	return itm, nil
+}
+
+func (c Channels) DeleteMessage(
+	ctx context.Context,
+	teamID, itemID, containerID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(itemID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "deleting mail message")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// replies
+// ---------------------------------------------------------------------------
+
+// GetReplies retrieves a Messageable item.
+func (c Channels) GetReplies(
+	ctx context.Context,
+	teamID, channelID, itemID string,
+) (serialization.Parsable, error) {
+	replies, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(channelID).
+		Messages().
+		ByChatMessageId(itemID).
+		Replies().
+		Get(ctx, nil)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return replies, nil
+}
+
+func (c Channels) PostReply(
+	ctx context.Context,
+	teamID, containerID, messageID string,
+	body models.ChatMessageable,
+) (models.ChatMessageable, error) {
+	itm, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(messageID).
+		Replies().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating reply message")
+	}
+
+	if itm == nil {
+		return nil, clues.New("nil response reply to message creation").WithClues(ctx)
+	}
+
+	return itm, nil
+}
+
+func (c Channels) DeleteReply(
+	ctx context.Context,
+	teamID, itemID, containerID, replyID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages().
+		ByChatMessageId(itemID).
+		Replies().
+		ByChatMessageId1(replyID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Wrap(ctx, err, "deleting mail message")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// Helpers
+// ---------------------------------------------------------------------------
+
+func MessageInfo(msg models.ChatMessageable, size int64) *details.GroupsInfo {
```

Comment:
suggestion: namespacing this func helps reduce collision between this and other types of "messages".  The other funcs are methods on the Channel client, so they're already namespaced.
```suggestion
func ChannelMessageInfo(msg models.ChatMessageable, size int64) *details.GroupsInfo {
```